recipes/root/all/conanfile.py

```diff
@@ -0,0 +1,210 @@
+    topics = ("data-analysis", "physics")
+    settings = ("os", "compiler", "build_type", "arch")
+    options = {
+        # Don't allow static build as it is not supported
+        # see: https://sft.its.cern.ch/jira/browse/ROOT-6446
+        # TODO: shared option should be reinstated when hooks issue is resolved
+        # (see: https://github.com/conan-io/hooks/issues/252)
+        # "shared": [True],
+        "fPIC": [True, False],
+        "python": PythonOption.ALL,
+    }
+    default_options = {
+        # "shared": True,
+        "fPIC": True,
+        "libxml2:shared": True,
+        "sqlite3:shared": True,
+        # default python=off as there is currently no libpython in Conan center
+        "python": PythonOption.OFF,
+    }
+    generators = ("cmake_find_package",)
+    requires = (
+        "opengl/system",
+        "libxml2/2.9.10",
+        "glu/system",
+        "xorg/system",
+        "sqlite3/3.33.0",
+        "libjpeg/9d",
+        "libpng/1.6.37",
+        "libcurl/7.73.0",
+        # "libuuid/1.0.3",
+    )
+
+    @property
+    def _rootsrcdir(self) -> str:
+        version = self.version.replace("v", "")
+        return f"root-{version}"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        # Patch ROOT CMake to use Conan SQLITE
+        tools.replace_in_file(
+            f"{self._rootsrcdir}{os.sep}CMakeLists.txt",
+            "project(ROOT)",
+            """project(ROOT)
+            find_package(SQLite3 REQUIRED)
+            set(SQLITE_INCLUDE_DIR ${SQLITE3_INCLUDE_DIRS})
+            set(SQLITE_LIBRARIES SQLite::SQLite)
+            """,
+        )
+        # Fix execute permissions on scripts
+        scripts = [
+            filename
+            for pattern in (
+                f"**{os.sep}configure",
+                f"**{os.sep}*.sh",
+                f"**{os.sep}*.csh",
+                f"**{os.sep}*.bat",
+            )
+            for filename in glob(pattern, recursive=True)
+        ]
+        for s in scripts:
+            self._make_file_executable(s)
+
+    def _make_file_executable(self, filename):
+        st = os.stat(filename)
+        os.chmod(filename, st.st_mode | stat.S_IEXEC)
+
+    @contextmanager
+    def _configure_cmake(self) -> CMake:
+        cmake = CMake(self)
+        version = self.version.replace("v", "")
+        cmakelibpath = ";".join(self.deps_cpp_info.lib_paths)
+        cmakeincludepath = ";".join(self.deps_cpp_info.include_paths)
+        cmake.configure(
+            source_folder=f"root-{version}",
+            defs={
+                # TODO: Remove BUILD_SHARED_LIBS option when hooks issue is resolved
+                # (see: https://github.com/conan-io/hooks/issues/252)
+                "BUILD_SHARED_LIBS": "ON",
+                "fail-on-missing": "ON",
+                "CMAKE_CXX_STANDARD": self._CMAKE_CXX_STANDARD,
+                # Prefer builtins where available
```

Comment:
typically as a CCI consumer, having the flexibility to choose and manage my deps is very important.

Personally I don't like seeing so many so many common and widely available packages because it will just duplicate the headache of managing everything

I would like to strongly encourage you to use the CCI packages where available

Reply:
The advantage of going with the builtins is that (in theory) you are guaranteed to have a set that work together.

But I see your point. I will look at adding in dependencies from CCI where they are available.

Reply:
I switched to using CCI packages where they are available in 7f503ea.